cli/build.go

```diff
@@ -237,17 +226,36 @@ func (cmd BuildFlags) getBuildPlan(
 // If -load is specified, will load the image into the local docker daemon.
 func (cmd BuildFlags) Build(contextDir string) error {
 	log.Infof("Starting Makisu build (version=%s)", utils.BuildHash)
+
 	imageName, err := cmd.getTargetImageName()
 	if err != nil {
 		return fmt.Errorf("failed to get target image name: %v", err)
 	}
 
-	buildPlan, err := cmd.getBuildPlan(contextDir, imageName)
+	// Convert context dir to absolute path.
+	contextDir, err = filepath.Abs(contextDir)
 	if err != nil {
-		return fmt.Errorf("failed to create build plan: %s", err)
+		return fmt.Errorf("failed to resolve context dir: %s", err)
+	}
+
+	// Create BuildContext.
+	buildContext, err := context.NewBuildContext("/", contextDir, cmd.imageStore)
+	if err != nil {
+		return fmt.Errorf("failed to create initial build context: %s", err)
+	}
+	defer buildContext.Cleanup()
```

Comment:
I think we should not cleanup the buildContext here. This would essentially make sure we leave `makisu-storage/stages/<stage id>` in there for debugging stages.

Reply:
I prefer to keep existing behavior until we have a way to actually debug that.